test/unit/test-amp-story-player.js

```diff
@@ -385,6 +389,90 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       );
     });
 
+    it('adds stories programmatically', async () => {
+      buildStoryPlayer(3);
+      await manager.loadPlayers();
+
+      const storyObjects = createStoryObjects(1);
+      playerEl.add(storyObjects);
+
+      const stories = toArray(playerEl.querySelectorAll('a'));
+
+      expect(stories.length).to.eql(4);
```

Comment:
Does this still work now that we're not appending the anchor to the HTML?

Reply:
Great catch! Given there is no other public exposure of the stories within the player, would it be okay if I put back the anchor appending when adding stories? This can be addressed in the follow up as well when we get rid of the HTMLAnchors

Reply:
I think we would rather check the `this.stories_` private property of the player here instead of doing that. We could add a `getStories` method that exposes this property as well.

Reply:
As discussed in our sync, I went with `getStories`, which was added in the latest commits

Reply:
I already LGTM'd this PR so feel free to ignore my comment, but we should not expose a `getStories` public method.
Multiple reasons:

1. The fact that it returns an array of <a> tags does not make sense, it's extremely counter intuitive
2. We'll migrate away from `stories_` being an array of anchors to an array of objects, which fixes (1), but is a breaking change
3. Making something public for testing reasons is a bad testing practice (cf go/unit-test-practices?polyglot=java#public-apis-dos-and-donts)

Reply:
@gmajoulet since we're already exposing a method that modifies the internal array, don't you think publishers might find it useful access to this array? 

If so, I think we can expose it for them to use. We can also specify in the docs what the expected structure will look like once we define this.

Reply:
We should only expose the array if it returns the expected output.
To unblock this PR, I'd be okay with `getStories()` looping over the anchors to return an array of objects that have the format we want to use when we refactor the `stories_` array.

Reply:
Done in the latest commit!